lib/web/apiserver.go

```diff
@@ -891,6 +893,10 @@ func (h *Handler) bindDefaultEndpoints() {
 	// Returns logins included in the Connect My Computer role of the user.
 	// Returns an empty list of logins if the user does not have a Connect My Computer role assigned.
 	h.GET("/webapi/connectmycomputer/logins", h.WithAuth(h.connectMyComputerLoginsList))
+
+	// Implements the agent version server.
+	// Channel can contain "/", hence the use of a catch-all parameter
+	h.GET("/webapi/automaticupgrades/*request", h.WithUnauthenticatedHighLimiter(h.automaticUpgrades))
```

Comment:
To wrap up automatic upgrades, we are also planning to support auto upgrades for Teleport client tools (tsh/tctl) and we wanted to be able to serve a separate version for client tools and agents.

With this design we can probably serve both versions by accepting an additional parameter to identify client tools vs agent. Alternatively we can have separate endpoints something like `automaticupgrades/agent/...` and `automaticupgrades/tools/...`. I think either design should work, but not sure which would be better.

Reply:
I think we have 3 approaches:
- use a different base path: `/automaticupgrades/<channel name>` vs `/agentautomatricupgrades/<channel name>`
- create different channels, one for agent and one for tool `/stable/cloud/agent` and `/stable/cloud/tool` (I'd also keep `/stable/cloud` for backward compact). This is a bit hacky but requires no development.
- expand the channel to contain 2 versions: agent and tool (and maybe default to both being equal). For example `/automaticupgrades/stable/cloud/version` and `/automaticupgrades/stable/cloud/agentVersion`

I'm not sure either which is best. Maybe @fspmarshall will have an opinion. I'll sleep on this and will come back to you on Monday.

Reply:
Of the proposals listed I'd probably lean toward just using different channel names since that seems dead simple (and leaves the option open for pointing agents and tools at the same channel if you're feelin' lazy), but I don't have a strong opinion either way.  I haven't been involved in the tool upgrade discussions, so I'm not sure whether or not it makes sense to lock them into having the same format/featureset.